docs/dev/design/file-tree-diff.rst

```diff
@@ -124,19 +124,27 @@ Storing results
 
 Doing a diff between two versions can be expensive, so we need to store the results.
 
-We can store the results in the DB (``VersionDiff``), or in S3 (``diff/project/version-a/version-b.json``), or maybe a combination of both.
-The information to store would contain some information about the versions compared, the build, and the diff itself.
+We can store the results in the DB (``VersionDiff``).
+The information to store would contain some information about the versions compared, the builds, and the diff itself.
+
+.. code:: python
+
+   class VersionDiff(models.Model):
+       version_a = models.ForeignKey(Version, on_delete=models.CASCADE, related_name='diff_a')
+       version_b = models.ForeignKey(Version, on_delete=models.CASCADE, related_name='diff_b')
+       build_a = models.ForeignKey(Build, on_delete=models.CASCADE, related_name='diff_a')
+       build_b = models.ForeignKey(Build, on_delete=models.CASCADE, related_name='diff_b')
```

Comment:
I do wonder if this should be a build object, or just a commit hash? I guess they work similarly, but there could be multiple different versions on the same commit, so it would be nice to not have to duplicate that work. However, we can also pretty easily get the commit via the Build.

Not 100% sure what is best here, but perhaps something to consider?

Reply:
A build can be triggered for the same commit and still change the output, say it takes into consideration the time or an env variable.